Sources/MongoSwift/BSON/BsonDecoder.swift

```diff
@@ -576,18 +576,12 @@ private extension DecodingError {
         let description = "Expected to find a value that can be represented as a \(expectation), " +
                          "but found value \(String(describing: reality)) of type \(type(of: reality)) instead."
         return .typeMismatch(expectation, Context(codingPath: path, debugDescription: description))
-
     }
 }
 
 /// This needs to be in this file to access some fileprivate decoder properties
 extension Document: Decodable {
 
-    static let decodeToTypes: [Decodable.Type] = [Double.self, String.self, Binary.self, ObjectId.self,
```

Comment:
accidentally left this in the decoder while testing your proposed approach, oops